src/coreComponents/constitutive/permeability/DisplacementDependentPermeability.hpp

```diff
@@ -0,0 +1,165 @@
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file DisplacementDependentPermeability.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_PERMEABILITY_DISPLACEMENTDEPENDENTPERMEABILITY_HPP_
+#define GEOSX_CONSTITUTIVE_PERMEABILITY_DISPLACEMENTDEPENDENTPERMEABILITY_HPP_
+
+#include "constitutive/permeability/PermeabilityBase.hpp"
+
+
+namespace geosx
+{
+namespace constitutive
+{
+
+class DisplacementDependentPermeabilityUpdate : public PermeabilityBaseUpdate
+{
+public:
+
+  DisplacementDependentPermeabilityUpdate( arrayView3d< real64 > const & permeability,
+                                           arrayView3d< real64 > const & dPerm_dPressure,
+                                           arrayView3d< real64 > const & dPerm_dDispJump,
+                                           real64 const shearDispThreshold,
+                                           real64 const maxPermMultiplier,
+                                           arrayView3d< real64 > const & iniPermeability )
+    : PermeabilityBaseUpdate( permeability, dPerm_dPressure ),
+    m_dPerm_dDispJump( dPerm_dDispJump ),
+    m_shearDispThreshold( shearDispThreshold ),
+    m_maxPermMultiplier( maxPermMultiplier ),
+    m_iniPermeability( iniPermeability )
+  {}
+
+  GEOSX_HOST_DEVICE
+  void compute( real64 const ( &dispJump )[3],
+                arraySlice1d< real64 > const & permeability,
+                arraySlice1d< real64 > const & dPerm_dDispJump ) const;
+
+  GEOSX_HOST_DEVICE
+  virtual void updateFromShearDisplacement( localIndex const k,
+                                            localIndex const q,
+                                            real64 const ( &dispJump )[3] ) const override
+  {
+    compute( dispJump,
+             m_permeability[k][q],
+             m_dPerm_dDispJump[k][q] );
+  }
+
+private:
+
+  /// dPermeability_dDispJump
+  arrayView3d< real64 > m_dPerm_dDispJump;
+
+  /// Threshold of shear displacement
+  real64 m_shearDispThreshold;
+
+  /// Maximum permeability multiplier
+  real64 m_maxPermMultiplier;
+
+  /// Initial permeability tensor
+  arrayView3d< real64 > m_iniPermeability;
+
+};
+
+
+class DisplacementDependentPermeability : public PermeabilityBase
+{
+public:
+
+  DisplacementDependentPermeability( string const & name, Group * const parent );
+
+  std::unique_ptr< ConstitutiveBase > deliverClone( string const & name,
+                                                    Group * const parent ) const override;
+
+  virtual void allocateConstitutiveData( dataRepository::Group & parent,
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  static string catalogName() { return "DisplacementDependentPermeability"; }
```

Comment:
Displacement dependent is quite generic. I would propose to specify this as slip dependent or fracture shear displacement dependent, etc.